hollow/src/main/java/com/netflix/hollow/api/producer/validation/DuplicateDataDetectionValidator.java

```diff
@@ -85,23 +102,42 @@ private String getDuplicateIDsString(Collection<Object[]> dupKeysCollection) {
         return message.toString();
 	}
 
-	private PrimaryKey getPrimaryKey(ReadState readState) {
+	private PrimaryKey getPrimaryKey(ReadState readState, Builder statusBuilder) {
 		PrimaryKey primaryKey = null;
 
 		if (fieldPathNames == null) {
 			HollowSchema schema = readState.getStateEngine().getSchema(dataTypeName);
 			if (schema.getSchemaType() != (SchemaType.OBJECT))
-				throw new ValidationException("Primary key validation is defined but schema type of "+ dataTypeName+" is not Object. This validation cannot be done.");
+				handleEndValidation(statusBuilder, Status.FAIL,  String.format(NOT_AN_OBJECT_ERROR_MSGR_FORMAT, dataTypeName));
+			
 			HollowObjectSchema oSchema = (HollowObjectSchema) schema;
 			primaryKey = oSchema.getPrimaryKey();
 		} else {
 			primaryKey = new PrimaryKey(dataTypeName, fieldPathNames);
 		}
 		if (primaryKey == null)
-			throw new ValidationException(
-					"Primary key validation defined but unable to find primary key for data type " + dataTypeName);
+			handleEndValidation(statusBuilder, Status.FAIL,  String.format(NO_PRIMARY_KEY_ERRRO_MSG_FORMAT, dataTypeName));
 
 		return primaryKey;
 	}
-
+	
+	private void handleEndValidation(Builder statusBuilder, Status status, String message) {
+		statusBuilder.withStatus(status);
+		if(message != null){
```

Comment:
Wonder if this can be improved since without context and just reading this code it feels a bit odd to throw exception if message is not null without checking on the status.    E.g. status == SUCCESS and message != null.

Reply:
Ok.

Reply:
Done.